arangod/Replication2/ReplicatedLog/NetworkAttachedFollower.cpp

```diff
@@ -0,0 +1,67 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2020-2021 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Tobias Gödderz
+////////////////////////////////////////////////////////////////////////////////
+
+#include "NetworkAttachedFollower.h"
+
+#include <utility>
+
+#include <Network/ConnectionPool.h>
+#include <Network/Methods.h>
+
+#include "Replication2/ReplicatedLog/NetworkMessages.h"
+
+using namespace arangodb;
+using namespace arangodb::replication2;
+using namespace arangodb::replication2::replicated_log;
+
+NetworkAttachedFollower::NetworkAttachedFollower(network::ConnectionPool* pool,
+                                                 ParticipantId id,
+                                                 DatabaseID database, LogId logId)
+    : pool(pool), id(std::move(id)), database(std::move(database)), logId(logId) {}
+
+auto NetworkAttachedFollower::getParticipantId() const noexcept -> ParticipantId const& {
+  return id;
+}
+
+auto NetworkAttachedFollower::appendEntries(AppendEntriesRequest request)
+    -> futures::Future<AppendEntriesResult> {
+  VPackBufferUInt8 buffer;
+  {
+    VPackBuilder builder(buffer);
+    request.toVelocyPack(builder);
+  }
+
+  auto path = "_api/log/" + std::to_string(logId.id()) + "/appendEntries";
+
+  network::RequestOptions opts;
+  opts.database = database;
+  auto f = network::sendRequest(pool, "server:" + id, arangodb::fuerte::RestVerb::Post,
```

Comment:
This will use the default timeout (120s ?). Not sure if we want to give up earlier or later. Should there be an explicit, shorter or longer timeout set here?

Reply:
🤔 This depends on what the practical implications of a 120s timeout are. If it is just a simple reconnect-resend loop that only returns an error if the message was sent completely, then this should be fine. 

In the end the ReplicatedLog system will simply try again with exponential back-off (limited to ~10s) forever.

Reply:
Afaik fuerte will retry (on certain network errors) until the timeout is reached. That should be fine.

The only thing that might be improved by a shorter timeout is that we might be able to squeeze some additional log entries in this message, but I don't think it can increase the throughput noticeably.